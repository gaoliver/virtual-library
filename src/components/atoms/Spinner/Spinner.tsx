import React from 'react';
import {Platform} from 'react-native';
import {ISpinnerProps, Spinner as NBSpinner} from 'native-base';
import {colors} from '@/theme/colors';

export const Spinner = ({...props}: ISpinnerProps) => {
  return (
    <NBSpinner
      {...props}
      {...(Platform.OS === 'android' && {color: colors.secondary})}
    />
  );
};
