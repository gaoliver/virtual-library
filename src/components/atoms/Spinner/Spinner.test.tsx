import React from 'react';
import renderer from 'react-test-renderer';
import {it, expect, jest} from '@jest/globals';
import {Spinner} from './Spinner';
import {NativeBaseProvider} from 'native-base';

it('Spinner renders correctly', () => {
  jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon');

  const tree = renderer
    .create(
      <NativeBaseProvider>
        <Spinner />
      </NativeBaseProvider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
