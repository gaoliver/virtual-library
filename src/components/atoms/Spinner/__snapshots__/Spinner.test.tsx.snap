// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
